### PR TITLE
feat: new ci base on apisix-runtime with specified version

### LINKED ISSUE #10401: Proposal: allow break change for APISIX-base package
### Description

The current problem is that APISIX limits dependence on the APISIX-base version during the packaging process, which results in all historical versions being affected when the next APISIX-base version makes destructive updates.
Detailed description: The current version 3.5 APISIX specifies the APISIX-base version in the package. The minimum version is 1.21.4.1.7 and the maximum is 1.21.5. If the next apisix-base version is 1.21.4.1.8, it is a destructive update. The released apisix is affected. Because installing APISIX will look for the latest version within the version range.

```shell
# Determine the min version of openresty or apisix-base
if [ "$OPENRESTY" == "apisix-base" ]
then
     min_or_version="1.21.4.1.7"
     max_or_version="1.21.5"
elif [ "$OPENRESTY" == "apisix-base-latest" ]
then
fpm -f -s dir -t "$PACKAGE_TYPE" \
     --"$PACKAGE_TYPE"-dist "$dist" \
     -n "$artifact" \
     -a "$(uname -i)" \
     -v "$PACKAGE_VERSION" \
     --iteration "$ITERATION" \
     -d "$OPENRESTY >= $min_or_version" \ 1.x
     -d "$OPENRESTY < $max_or_version" \ 2.x
```

The current idea is to solve the problem by adjusting version dependencies. The current situation is that the version number of apisix-base is based on the Openresty version number + patch number. For example, apisix-base-1.21.4.1.8 means that this is based on Openresty 1.21.4.1 8 patches.

Solution overview: Change the package name to **APISIX-runtime**; add the own version of APISIX runtime. The version information of openresty is no longer included because the APISIX version is more radical than openresty; the nginx version information is not included to weaken nginx.

For example: `APISIX-runtime.MAJOR.MINOR. PATCH`,  `APISIX-runtime` is the package name.
- MAJOR major version number: when you make incompatible API changes.
- MINOR minor version number: When you make a backward compatible functional addition.
- PATCH revision number: when you make a backwards compatibility issue fix.

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

Fixes #10401

1. `Centos` support for wasm test has been removed due to compatibility issues.
2. Replace the original `openresty-debug` of centos with apisix-runtime
3. The version of apisix-runtime is specified for CI

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

I have test this PR with https://github.com/Sn0rt/apisix/pull/21

### CODE REVIEW

File: .github/workflows/centos7-ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
           - t/plugin/[a-k]*
           - t/plugin/[l-z]*
           - t/admin t/cli t/config-center-yaml t/control t/core t/debug t/discovery t/error_page t/misc
-          - t/node t/pubsub t/router t/script t/secret t/stream-node t/utils t/wasm t/xds-library
```
Comment (monkeyDluffy6017) [inline .github/workflows/centos7-ci.yml:34]: Wasm didn't run on the current ci?

File: .github/workflows/centos7-ci.yml [hunk 2]
```diff
@@ -94,6 +94,14 @@ jobs:
         cd ..
         rm -rf $(ls -1 --ignore=apisix-build-tools --ignore=t --ignore=utils --ignore=ci --ignore=Makefile --ignore=rockspec)
 
+    - name: Start Dubbo Backend
+      run: |
+        sudo apt install -y maven
+        cd t/lib/dubbo-backend
+        mvn package
+        cd dubbo-backend-provider/target
+        java -Djava.net.preferIPv4Stack=true -jar dubbo-demo-provider.one-jar.jar > /tmp/java.log &
```
Comment (monkeyDluffy6017) [inline .github/workflows/centos7-ci.yml:103]: Why do you add this?

File: ci/centos7-ci.sh [hunk 3]
```diff
@@ -31,9 +32,19 @@ install_dependencies() {
...
+    source scl_source enable devtoolset-9
+    set -eu
+
     # install openresty to make apisix's rpm test work
     yum install -y yum-utils && yum-config-manager --add-repo https://openresty.org/package/centos/openresty.repo
-    yum install -y openresty-1.21.4.2 openresty-debug-1.21.4.2 openresty-openssl111-debug-devel pcre pcre-devel
+    wget "https://raw.githubusercontent.com/api7/apisix-build-tools/apisix-runtime/${APISIX_RUNTIME}/build-apisix-runtime-debug-centos7.sh"
+    wget "https://raw.githubusercontent.com/api7/apisix-build-tools/apisix-runtime/${APISIX_RUNTIME}/build-apisix-runtime.sh"
+    chmod +x build-apisix-runtime-debug-centos7.sh
+    chmod +x build-apisix-runtime.sh
```
Comment (AlinsRan) [inline ci/centos7-ci.sh:46]: Doesn't seem to need `build-apisix-runtime.sh` ？

Comment (Sn0rt) [reply ci/centos7-ci.sh:46]: build-apisix-runtime-debug-centos7.sh is called in this script

Comment (AlinsRan) [reply ci/centos7-ci.sh:46]: Oh, okay.

**Review by AlinsRan — APPROVED** [review]

File: .github/workflows/centos7-ci.yml (hunk 1 above)
Comment (monkeyDluffy6017) [reply .github/workflows/centos7-ci.yml:34]: The centos don't use the apisix-base before

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**